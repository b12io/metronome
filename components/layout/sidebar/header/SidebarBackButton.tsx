
import React from "react";

import { HEADER_BACK_ICON_COLOR } from "./constants.es6";
import { Back } from "../../../Icons.es6";

type Props = {
  onClick(): void;
  hideText: boolean;
  text: string;
};

class SidebarBackButton extends React.Component<Props> {

  backButtonRef = React.createRef<HTMLDivElement>();

  static defaultProps = {
    hideText: false,
    text: 'Back'
  };

  handleClick = () => {
    if (this.backButtonRef.current) {
      this.backButtonRef.current.focus();
    }
    this.props.onClick();
  };

  render() {
    return <div className="ds-sidebar__header-back-button-container" onClick={this.handleClick} ref={this.backButtonRef} tabIndex="-1">
        <button type="button">
          <Back color={HEADER_BACK_ICON_COLOR} />
          {!this.props.hideText && <span className="ds-sidebar__header-back-text">{this.props.text}</span>}
        </button>
      </div>;
  }
}

export default SidebarBackButton;