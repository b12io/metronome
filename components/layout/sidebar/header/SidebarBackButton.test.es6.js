/* global expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SidebarBackButton from './SidebarBackButton.es6.js'

describe('SidebarBackButton component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<SidebarBackButton onClick={jest.fn()} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('fires onClick prop when clicked', () => {
    const wrapper = shallow(<SidebarBackButton onClick={jest.fn()} />)
    wrapper.find('.ds-sidebar__header-back-button-container').simulate('click')
    expect(wrapper.instance().props.onClick).toHaveBeenCalled()
  })
})
