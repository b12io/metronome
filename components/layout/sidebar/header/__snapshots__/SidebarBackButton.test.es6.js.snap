// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarBackButton component renders correctly 1`] = `
<div
  className="ds-sidebar__header-back-button-container"
  onClick={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
>
  <button
    type="button"
  >
    <svg
      className=""
      height={16}
      solid="true"
      version="1.1"
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        fill="#84839C"
        points="14.85 6.5625 14.85 8.4375 3.6 8.4375 8.75625 13.59375 7.425 14.925 0 7.5 7.425 0.075 8.75625 1.40625 3.6 6.5625"
      />
    </svg>
    <span
      className="ds-sidebar__header-back-text"
    >
      Back
    </span>
  </button>
</div>
`;
